### PR TITLE
More module level documentation

### PR DESCRIPTION
**Pull request by erikbern**

### CODE REVIEW

**Comment by erikbern** [comment]
Btw I accidentally pushed this to master first., not sure if you saw it. I reverted it because it had a pep8 issue.

**Comment by erikbern** [comment]
I've been working on documentation since it's such a low-cost thing with high impact. Documentation is actually starting to get useful now that it's split up and there's some more stuff. Will add some more soon

File: luigi/contrib/sqla.py [hunk 1]
```diff
@@ -141,8 +141,14 @@ def requires(self):
...
 import luigi
-import sqlalchemy
 import os
+import warnings
+
+try:
+    import sqlalchemy
+except ImportError:
+    # Don't fail on import because we want the documentation to be generated
+    warnings.warn("sqlalchemy could not be imported, db_task_history will not work", warnings.ImportWarning, stacklevel=3)
```
Comment (Tarrasch) [inline luigi/contrib/sqla.py]: This doesn't seem related to docs? :)

Comment (erikbern) [inline luigi/contrib/sqla.py]: sphinx-apidoc needs to be able to import the module to generate docs. Currently `luigi.contrib.sqla` throws an exception at the module level and that's why no documentation is generated: http://luigi.readthedocs.org/en/latest/api/luigi.contrib.sqla.html

Comment (erikbern) [inline luigi/contrib/sqla.py]: Also see here: https://readthedocs.org/builds/luigi/2375884/

```
/var/build/user_builds/luigi/checkouts/latest/doc/api/luigi.contrib.sqla.rst:4: WARNING: autodoc: failed to import module u'luigi.contrib.sqla'; the following exception was raised:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/luigi/envs/latest/local/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 335, in import_object
    __import__(self.modname)
  File "/var/build/user_builds/luigi/checkouts/latest/luigi/contrib/sqla.py", line 144, in <module>
    import sqlalchemy
ImportError: No module named sqlalchemy
```

Comment (gpoulin) [inline luigi/contrib/sqla.py]: It also fails for db_task_history for the same reason.

Comment (erikbern) [inline luigi/contrib/sqla.py]: Yeah, I fixed both
